frame/collective/src/lib.rs

```diff
@@ -631,6 +631,7 @@ mod tests {
 
 			System::set_block_number(4);
 			assert_ok!(Collective::close(Origin::signed(4), hash.clone(), 0));
+			System::finalize();
```

Comment:
we really need a better abstraction for advancing block in tests

Reply:
Should probably use the same abstraction as below: 

```rust 
fn assert_events(events) {
    // check if system is finalized(), if not do it
    // assert events
}

Reply:
Refactored, now just use `collect_events` instead of `events` (new name also implies that it is a mutable operation)